src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -542,6 +542,25 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
     })
   }
 
+  override def acceptTermsOfService(userId: WorkbenchUserId, tosVersion: String, samRequestContext: SamRequestContext): IO[Boolean] = {
+    serializableWriteTransaction("acceptTermsOfService", samRequestContext)({ implicit session =>
+      val u = UserTable.column
+      samsql"""update ${UserTable.table} set ${u.acceptedTosVersion} = ${tosVersion}
+              where ${u.id} = ${userId}
+              and (${u.acceptedTosVersion} is null
+                or ${u.acceptedTosVersion} != ${tosVersion})""".update().apply() > 0
+    })
+  }
+
+  override def rejectTermsOfService(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Boolean] = {
+    serializableWriteTransaction("rejectTermsOfService", samRequestContext)({ implicit session =>
+      val u = UserTable.column
+      samsql"""update ${UserTable.table} set ${u.acceptedTosVersion} = null
+              where ${u.id} = ${userId}
+              and ${u.acceptedTosVersion} is not null""".update().apply() > 0
```

Comment:
same/similar question here I suppose, why check that the ToS has been accepted? Are we checking upstream that a user isn't double rejecting?